### PR TITLE
Add uclamp support improvements

### PR DESCRIPTION
**Pull request by qais-yousef**
Some improvements on top of #81 which addresses some review comments and gets rid of optional building with sched-deadline which is something that creeped up while discussing the PR. It relies on #81 being merged first as it builds on top.

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -1,6 +1,2 @@
-if SET_DLSCHED
```
Comment (vingu-linaro) [inline Makefile.am]: This can't be removed unconditionally.
We build libdl only if libc doesn't support sched_setattr.

File: configure.ac [hunk 2]
```diff
@@ -18,14 +18,6 @@ AC_CHECK_LIB([rt], [clock_gettime])
 AC_CHECK_LIB([json-c], [json_object_from_file], [], [AC_MSG_ERROR([json-c libraries required])])
 AC_CHECK_FUNCS(sched_setattr, [], [])
 
-AC_ARG_WITH([deadline],
-	[AS_HELP_STRING([--with-deadline], 
-		[Add support for SCHED_DEADLINE])],
-	[AC_DEFINE([DLSCHED], [1], [Define if you have SCHED_DEADLINE support])], 
-	[with_deadline=no])
-
-AM_CONDITIONAL([SET_DLSCHED], [test x$with_deadline != xno && test !HAVE_SCHED_SETATTR])
```
Comment (vingu-linaro) [inline configure.ac]: this should be kept

File: src/rt-app.c [hunk 3]
```diff
@@ -1018,6 +1001,13 @@ static void set_thread_priority(thread_data_t *data, sched_data_t *sched_data)
 				   sched_data->util_max);
 		}
 
+		/*
+		 * we can't change clamp values without valid policy and
+		 * prioroity values.
```
Comment (vingu-linaro) [inline src/rt-app.c]: s/prioroity/priority/

File: src/rt-app.c [hunk 4]
```diff
@@ -1018,6 +1001,13 @@ static void set_thread_priority(thread_data_t *data, sched_data_t *sched_data)
 				   sched_data->util_max);
 		}
 
+		/*
+		 * we can't change clamp values without valid policy and
```
Comment (vingu-linaro) [inline src/rt-app.c]: uppercase: We

File: Makefile.am (hunk 1 above)
Comment (qais-yousef) [reply Makefile.am]: I remember in our call when Patrick was here we agreed to always compile deadline in as it's been around for long enough now. Am I mis-remembering?

File: src/rt-app.c (hunk 3 above)
Comment (qais-yousef) [reply src/rt-app.c]: done

File: src/rt-app.c (hunk 4 above)
Comment (qais-yousef) [reply src/rt-app.c]: done

File: Makefile.am (hunk 1 above)
Comment (vingu-linaro) [reply Makefile.am]: Yes it's right, you don't mis-remember but SET_DLSCHED is not about compiling the support of deadline scheduler but about compiling a local support of sched_setattr if the lib doesn't support it.
So DLSCHED must be removed but not SET_DLSCHED

Comment (qais-yousef) [reply Makefile.am]: Ah I see. Okay let me see if I can hack something and updated the PR

Comment (vingu-linaro) [reply Makefile.am]: It should be just keeping this : if SET_DLSCHED in Makefile.am
as well as the AM_CONDITIONAL([SET_DLSCHED] .... in configure.ac

Comment (qais-yousef) [reply Makefile.am]: Okay thanks. Done that. Just to confirm, all the `#ifdef DLSCHED` in the *.[ch] files should be removed now, right?

Comment (qais-yousef) [reply Makefile.am]: And the configure.ac test is now shorter

Comment (vingu-linaro) [reply Makefile.am]: > Okay thanks. Done that. Just to confirm, all the `#ifdef DLSCHED` in the *.[ch] files should be removed now, right?
Yes.

**Comment by qais-yousef** [comment]
Thanks for the review @vingu-linaro. If there are no more reviewers intending to look at this it'd be nice if we can merge #81 then this one.

**Pull request merged**